akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala

```diff
@@ -256,4 +261,14 @@ object EventSourcedBehavior {
    * If not defined, the default `akka.persistence.typed.stash-capacity` will be used.
    */
   def withStashCapacity(size: Int): EventSourcedBehavior[Command, Event, State]
+
+  /**
+   * Invoke this callback when an event from another replica arrives, delaying persisting the event until the returned
+   * future completes, if the future fails the actor is crashed.
+   *
+   * Only used when the entity is replicated.
+   */
+  @ApiMayChange
+  def withReplicatedEventInterceptor(
```

Comment:
would have been nice if we could place this in the `ReplicatedEventSourcing`? would then require additional overloaded factory methods there?

Reply:
Yeah, would have to duplicate the builder but maybe worth it because a bit weird with these methods up here. It's anyway already in three places.

Reply:
One curious thing is that none of the Java API changes show up in MiMa, isn't adding a method to a superclass both source and binary incompatible?

Reply:
https://github.com/akka/akka/pull/32526/commits/9a2d2e5f11c1e90d9217f2ab6e48522c1012bc7a changed it for javadsl, but for scaladsl we still have to have it in the EventSourcedBehavior?

Reply:
Yeah for Scala we don't have an explicit protocol, the RESB is defined using the ESB API.

Reply:
Yes, I thought we could add overloaded factory methods in ReplicatedEventSourcing, and make the `withReplicatedEventInterceptor` internal in ESB. Maybe not important.

Reply:
The problem is that those factories are only for direct-db-replication, the "other" API is over in Akka Projection gRPC, so we'd need to implement it in those two places somehow, with internal API to set it. Not worth it IMO